feature-libs/organization/administration/core/guards/admin.guard.ts

```diff
@@ -15,17 +15,33 @@ import {
 import { UserAccountFacade } from '@spartacus/user/account/root';
 import { Observable } from 'rxjs';
 import { filter, map, pluck } from 'rxjs/operators';
+import { B2BUserService } from '../services';
+import { Location } from '@angular/common';
 
 @Injectable({
   providedIn: 'root',
 })
-export class AdminGuard implements CanActivate {
+export class AdminGuard implements CanActivate, CanMatch {
   constructor(
     protected userAccountFacade: UserAccountFacade,
     protected routingService: RoutingService,
-    protected globalMessageService: GlobalMessageService
+    protected globalMessageService: GlobalMessageService,
+    protected b2bUserService: B2BUserService,
+    protected location: Location
   ) {}
 
+  canMatch(): boolean {
+    const promise = this.b2bUserService.isUpdatingUserAllowed();
+    if (!promise) {
+      this.routingService.go({ cxRoute: 'organization' });
+      this.globalMessageService.add(
+        { key: 'organization.notification.notExist' },
+        GlobalMessageType.MSG_TYPE_WARNING
+      );
+    }
+    return promise;
```

Comment:
When redirecting to another URL inside a guard, it's recommended to return the `UrlTree` of the redirected URL, instead of returning `false` and triggering a new navigation.

```ts
    const isUpdatingUserAllowed = this.b2bUserService.isUpdatingUserAllowed();
    if (isUpdatingUserAllowed) {
      this.globalMessageService.add(
        { key: 'organization.notification.notExist' },
        GlobalMessageType.MSG_TYPE_WARNING
      );
      return this.router.parseUrl(
        this.semanticPathService.get('organization') ?? ''
      );
    }
    return true;
```

Reply:
done